src/cmds/qstat.c

```diff
@@ -2852,13 +2852,12 @@ qstat -B [-f] [-F format] [-D delim] [ server_name... ]\n";
 						any_failed = pbs_errno;
 					}
 				} else {
-				    /* check the server attribute max_job_sequence_id value */
-					check_seqid_len = check_max_job_sequence_id("qstat");
-					if (check_seqid_len == 1) {
-						how_opt |= ALT_DISPLAY_INCR_WIDTH; /* increase column width*/
-					} else if(check_seqid_len == -1) {
-						fprintf(stderr, "qstat: Unable to fetch the width format\n");
-						exit(1);
+					/* check the server attribute max_job_sequence_id value */
+					if (p_server != NULL) {
+						check_seqid_len = check_max_job_sequence_id(p_server);
```

Comment:
This is better, but will still call check_max_job_sequence_id() for every jobid, even though those jobs are from the same server. Thus, we should call this only once for one cnt2server() calls, way higher, where we do the pbs_statserver().